lib/matplotlib/testing/decorators.py

```diff
@@ -481,8 +482,8 @@ def _image_directories(func):
     ``$(pwd)/result_images/test_baz``.  The result directory is created if it
     doesn't exist.
     """
-    module_path = Path(sys.modules[func.__module__].__file__)
-    baseline_dir = module_path.parent / "baseline_images" / module_path.stem
+    module_path = Path(sys.modules[func.__module__].__file_)
+    baseline_dir = matplotlib_baseline_images.__file__ / "mpl_baseline_images" / module_path.stem
```

Comment:
consider falling back to the old behavior e.g. if `not func.__module__.startswith("matplotlib.", "mpl_toolkits."))` so that things keep working as previously for third-parties that may also be using this machinery.